tensorflow_addons/custom_ops/image/python/dense_image_warp_test.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for dense_image_warp."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import math
+
+import numpy as np
+
+import tensorflow as tf
+
+from tensorflow.python.framework import test_util as tf_test_util
+from tensorflow_addons.custom_ops.image.python import dense_image_warp as dense_image_warp_ops
+
+
+class DenseImageWarpTest(tf.test.TestCase):
+
+    def setUp(self):
+        np.random.seed(0)
+
+    @tf_test_util.run_all_in_graph_and_eager_modes
+    def test_interpolate_small_grid_ij(self):
+        grid = tf.constant(
+            [[0., 1., 2.], [3., 4., 5.], [6., 7., 8.]], shape=[1, 3, 3, 1])
+        query_points = tf.constant(
+            [[0., 0.], [1., 0.], [2., 0.5], [1.5, 1.5]], shape=[1, 4, 2])
+        expected_results = np.reshape(
+            np.array([0., 3., 6.5, 6.]), [1, 4, 1])
+
+        interp = dense_image_warp_ops._interpolate_bilinear(
```

Comment:
Oh, I don't realize there are some test cases for `_interpolate_bilinear`. Could you use `tf.function` decorator for `_interpolate_bilinear`? And paste the error logs, I'll find people to help resolve the issue.

Reply:
Here it is.
```
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //tensorflow_addons/custom_ops/image:dense_image_warp_test
-----------------------------------------------------------------------------
tail: unrecognized file system type 0x794c7630 for '/root/.cache/bazel/_bazel_root/c27e9809996ce9a9c0ed8dd79ef
0897b/execroot/__main__/bazel-out/k8-opt/testlogs/tensorflow_addons/custom_ops/image/dense_image_warp_test/tes
t.xml.log'. please report this to bug-coreutils@gnu.org. reverting to polling
Running tests under Python 2.7.6: /usr/bin/python
[ RUN      ] DenseImageWarpTest.test_gradients_exist
2019-02-27 14:34:18.291511: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instruction
s that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-02-27 14:34:18.311916: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 420000000
0 Hz
2019-02-27 14:34:18.312364: I tensorflow/compiler/xla/service/service.cc:162] XLA service 0x4033600 executing
computations on platform Host. Devices:
2019-02-27 14:34:18.312375: I tensorflow/compiler/xla/service/service.cc:169]   StreamExecutor device (0): <un
defined>, <undefined>
W0227 14:34:18.461117 139673821919040 deprecation.py:323] From /usr/local/lib/python2.7/dist-packages/tensorfl
ow/python/ops/array_grad.py:425: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be remo
ved in a future version.
Instructions for updating:
Use tf.cast instead.
[       OK ] DenseImageWarpTest.test_gradients_exist
[ RUN      ] DenseImageWarpTest.test_interpolate_small_grid_batched
[       OK ] DenseImageWarpTest.test_interpolate_small_grid_batched
[ RUN      ] DenseImageWarpTest.test_interpolate_small_grid_ij
[       OK ] DenseImageWarpTest.test_interpolate_small_grid_ij
[ RUN      ] DenseImageWarpTest.test_interpolate_small_grid_xy
[       OK ] DenseImageWarpTest.test_interpolate_small_grid_xy
[ RUN      ] DenseImageWarpTest.test_interpolation
2019-02-27 14:34:26.215331: W tensorflow/core/common_runtime/base_collective_executor.cc:214] BaseCollectiveEx
ecutor::StartAbort Invalid argument: Input to reshape is a tensor with 75 values, but the requested shape has
360
         [[{{node dense_image_warp/StatefulPartitionedCall/interpolate_bilinear/Reshape}}]]
2019-02-27 14:34:26.215363: E tensorflow/core/common_runtime/process_function_library_runtime.cc:764] Componen
t function execution failed: Invalid argument: Input to reshape is a tensor with 75 values, but the requested
shape has 360
         [[{{node dense_image_warp/StatefulPartitionedCall/interpolate_bilinear/Reshape}}]]
[  FAILED  ] DenseImageWarpTest.test_interpolation
[ RUN      ] DenseImageWarpTest.test_session
W0227 14:34:26.216089 139673821919040 deprecation.py:323] From /usr/lib/python2.7/contextlib.py:84: test_sessi
on (from tensorflow.python.framework.test_util) is deprecated and will be removed in a future version.
Instructions for updating:
Use `self.session()` or `self.cached_session()` instead.
[       OK ] DenseImageWarpTest.test_session
[ RUN      ] DenseImageWarpTest.test_size_exception
2019-02-27 14:34:26.216985: W tensorflow/core/common_runtime/base_collective_executor.cc:214] BaseCollectiveEx
ecutor::StartAbort Invalid argument: assertion failed: [Grid width must be at least 2.] [Condition x >= y did
not hold element-wise:x (interpolate_bilinear/strided_slice_2:0) = ] [1] [y (interpolate_bilinear/assert_great
er_equal_1/y:0) = ] [2]
         [[{{node dense_image_warp/StatefulPartitionedCall/interpolate_bilinear/assert_greater_equal_1/Assert/Assert}}]]
2019-02-27 14:34:26.217016: E tensorflow/core/common_runtime/process_function_library_runtime.cc:764] Componen
t function execution failed: Invalid argument: assertion failed: [Grid width must be at least 2.] [Condition x
 >= y did not hold element-wise:x (interpolate_bilinear/strided_slice_2:0) = ] [1] [y (interpolate_bilinear/as
sert_greater_equal_1/y:0) = ] [2]
         [[{{node dense_image_warp/StatefulPartitionedCall/interpolate_bilinear/assert_greater_equal_1/Assert/
Assert}}]]
[       OK ] DenseImageWarpTest.test_size_exception
[ RUN      ] DenseImageWarpTest.test_zero_flows
[       OK ] DenseImageWarpTest.test_zero_flows
======================================================================
ERROR: test_interpolation (__main__.DenseImageWarpTest)
test_interpolation (__main__.DenseImageWarpTest)
Apply _check_interpolation_correctness() for a few sizes and types.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/absl/third_party/unittest3_backport/case.py", line 37, in testP
artExecutor
    yield
  File "/usr/local/lib/python2.7/dist-packages/absl/third_party/unittest3_backport/case.py", line 162, in run
    testMethod()
  File "/root/.cache/bazel/_bazel_root/c27e9809996ce9a9c0ed8dd79ef0897b/execroot/__main__/bazel-out/k8-opt/bin
/tensorflow_addons/custom_ops/image/dense_image_warp_test.runfiles/__main__/tensorflow_addons/custom_ops/image
/python/dense_image_warp_test.py", line 190, in test_interpolation
    shape, im_type, flow_type)
  File "/root/.cache/bazel/_bazel_root/c27e9809996ce9a9c0ed8dd79ef0897b/execroot/__main__/bazel-out/k8-opt/bin
/tensorflow_addons/custom_ops/image/dense_image_warp_test.runfiles/__main__/tensorflow_addons/custom_ops/image
/python/dense_image_warp_test.py", line 165, in _check_interpolation_correctness
    flow=tf.convert_to_tensor(rand_flows))
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/eager/def_function.py", line 418, in __call__
    results = self._stateful_fn(*args, **kwds)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/eager/function.py", line 1256, in __call__
    return graph_function._filtered_call(args, kwargs)  # pylint: disable=protected-access
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/eager/function.py", line 542, in _filtered_ca
ll
    (t for t in nest.flatten((args, kwargs))
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/eager/function.py", line 595, in _call_flat
    outputs = self._inference_function.call(ctx, args)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/eager/function.py", line 383, in call
    ctx=ctx)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/eager/execute.py", line 66, in quick_execute
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "/usr/local/lib/python2.7/dist-packages/six.py", line 737, in raise_from
    raise value
InvalidArgumentError: Input to reshape is a tensor with 75 values, but the requested shape has 360
         [[{{node dense_image_warp/StatefulPartitionedCall/interpolate_bilinear/Reshape}}]] [Op:__inference_de
nse_image_warp_2318]

----------------------------------------------------------------------
Ran 8 tests in 9.152s

FAILED (errors=1)
```

Reply:
Can you push the change to github? Then we can reproduce the failure and debug it :-)

Reply:
Pushed 😃

Reply:
@karmel Hi, Karmel. Do you know the problem or can redirect to someone who does? Thanks.